Automattic/jetpack
# Add a Paid Newsletter Recommendation card

### Changes proposed in this Pull Request:
This PR adds a Recommendation to set up a Paid Newsletter. It triggers in the following conditions:-

- A site has the Newsletter module active (subscriptions)
- A site has over 100 subscribers but less than 100k subscribers
- It triggers as an action hooked to the fetch of the Subscriber stats (so needs the operator to visit the stats page).

### Jetpack product discussion

See: pbNhbs-8a3-p2

### Does this pull request change what data or activity we track or use?
No

### Testing instructions:
A pickle of a PR to test. It took me a while to get my test site to show as having more than 100 subscribers. I ended up having to manually add them from `https://wordpress.com/subscribers/{site}` importing via CSV they didn't seem to process on my ngrok site. 

- Get over 100 subscribers on your site 
- Clear the Jetpack Dev options 
- Visit the Dashboard there should be no recommendations shown
- Go to Stats
- Click the Subscribers tab - it should show you have over 100 subscribers
- Visit the Dashboard again - the red bubble should show up next to Recommendations
- Click the Recommendations tab - the prompt to set up a Paid Newsletter will show
- This then takes you through the flow of setting up a new plan. See the product discussion for a diagram of that flow.

It's worth nothing that on code review, the `total` in the current stats endpoint doesn't total the subscribers, it needs to pass `?type=all` to get around this I've summed the `total_wpcom` and `total_email` data to give the total which is the same with or without `?type=all` being passed. 

See: p1694436980423499/1694436172.515229-slack-C02TCEHP3HA for discussion.

### Here's how it should look

https://github.com/Automattic/jetpack/assets/4077604/6297e40a-89a9-42ec-992b-60e2726b117a
